docs/bigquery-usage.rst

```diff
@@ -236,3 +236,189 @@ Delete a table:
    >>> dataset = client.dataset('dataset_name')
    >>> table = dataset.table(name='person_ages')
    >>> table.delete()  # API request
+
+Jobs
+----
+
+Jobs describe actions peformed on data in BigQuery tables:
+
+- Load data into a table
+- Run a query against data in one or more tables
+- Extrat data from a table
+- Copy a table
+
+List jobs for a project:
+
+.. doctest::
+
+   >>> from gcloud import bigquery
+   >>> client = bigquery.Client()
+   >>> jobs = client.jobs()  # API request
+   >>> [(job.job_id, job.type, job.created, job.state) for job in jobs]
+   ['e3344fba-09df-4ae0-8337-fddee34b3840', 'insert', (datetime.datetime(2015, 7, 23, 9, 30, 20, 268260, tzinfo=<UTC>), 'done')]
+
+Querying data (synchronous)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Run a query which can be expected to complete within bounded time:
+
+.. doctest::
+
+   >>> from gcloud import bigquery
+   >>> client = bigquery.Client()
+   >>> query = """\
+   SELECT count(*) AS age_count FROM dataset_name.person_ages
+   """
+   >>> results = client.query(query, timeout_ms=1000)
+   >>> while not results.job_complete:
+   ...    time.sleep(10)
+   ...    results.reload()  # API request
+   >>> results.schema
+   [{'name': 'age_count', 'type': 'integer', 'mode': 'nullable'}]
+   >>> results.rows
+   [(15,)]
+
+.. note::
+   
+   If the query takse longer than the timeout allowd, ``results.job_complete``
+   will be False:  we therefore poll until it is completed.
+
+Querying data (asynchronous)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Background a query, loading the results into a table:
+
+.. doctest::
+
+   >>> from gcloud import bigquery
+   >>> client = bigquery.Client()
+   >>> query = """\
+   SELECT firstname + ' ' + last_name AS full_name,
+          FLOOR(DATEDIFF(CURRENT_DATE(), birth_date) / 365) AS age
+    FROM dataset_name.persons
+   """
+   >>> dataset = client.dataset('dataset_name')
+   >>> table = dataset.table(name='person_ages')
+   >>> job = client.query_async(query,
+   ...                          destination=table,
+   ...                          write_disposition='truncate')
+   >>> job.job_id
+   'e3344fba-09df-4ae0-8337-fddee34b3840'
+   >>> job.type
+   'load'
+   >>> job.created
+   None
+   >>> job.state
+   None
+
+.. note::
+
+   - ``gcloud.bigquery`` generates a UUID for each job.
+   - The ``created`` and ``state`` fields are not set until the job
+     is submitted to the BigQuery back-end.
+
+Then, begin executing the job on the server:
+
+.. doctest::
+
+   >>> job.submit()  # API call
+   >>> job.created
+   datetime.datetime(2015, 7, 23, 9, 30, 20, 268260, tzinfo=<UTC>)
+   >>> job.state
+   'running'
+
+Poll until the job is complete:
```

Comment:
Even if we had cross-Python support for it. it would complicate these examples needlessly.